credit_card_gateway.go

```diff
@@ -57,3 +62,30 @@ func (g *CreditCardGateway) Delete(ctx context.Context, card *CreditCard) error
 	}
 	return &invalidResponseError{resp}
 }
+
+//  ExpiringBetween Find list of credit card that expire between the specified dates
+func (g *CreditCardGateway) ExpiringBetween(ctx context.Context, fromDate, toDate time.Time) ([]*CreditCard, error) {
+	path := fmt.Sprintf("/payment_methods/all/expiring?start=%s&end=%s",
+		fromDate.Format("012006"),
+		toDate.Format("012006"))
+	resp, err := g.execute(ctx, "POST", path, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	type searchResult struct {
+		XMLName           xml.Name      `xml:"payment-methods"`
+		CurrentPageNumber int64         `xml:"current-page-number"`
+		PageSize          int64         `xml:"page-size"`
```

Comment:
How do we use the paging fields of this struct with the function? Or does this function only support the first page? If it only supports the first page, can we use the same pattern that was added in 55b62f92375c92ad5903deb2b4dc71065ccd3171 and using both the `/payment_methods/all/expiring_ids` and `/payment_methods/all/expiring` endpoints?  
Ref: [braintree/braintree_java](https://github.com/braintree/braintree_java/blob/2def3f6dcf3fbbfcb3c74d67716aadfa3f0658bd/src/main/java/com/braintreegateway/CreditCardGateway.java#L190-L207)

I think it'd also be worth adding the `Expired()` function too alongside `ExpiringBetween` since it should follow the same pattern.  
Ref: [braintree/braintree_java](https://github.com/braintree/braintree_java/blob/2def3f6dcf3fbbfcb3c74d67716aadfa3f0658bd/src/main/java/com/braintreegateway/CreditCardGateway.java#L164-L180)